lib/cloudant.js

```diff
@@ -540,11 +540,59 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
       return cb();
     }, function(err) {
       if (err) return cb(err);
-      mo.db.bulk({docs: docs}, function(err) {
+      mo.db.bulk({docs: docs}, function(err, result) {
         if (err) return cb(err);
-        var res = {};
-        res.count = docs.length;
-        return cb(err, res);
+        var errorArray = _.filter(result, 'error');
+        if (errorArray.length > 0) {
+          err = new Error(g.f(util.format('Unable to update 1 or more ' +
+          'document(s): %s', util.inspect(errorArray, 2))));
+          return cb(err);
+        } else {
+          return cb(err, {count: result.length});
+        }
+      });
+    });
+  });
+};
+
+/**
+ * Perform a bulk update on a model instance
+ *
+ * @param {String} model The model name
+ * @param {Array} datas List of data to be updated
+ * @callback {Function} cb The callback function
+ */
+Cloudant.prototype.massUpdate = function(model, datas, cb) {
+  debug('Cloudant.prototype.massUpdate %j %j',
+          datas);
+  var self = this;
+  var mo = self.selectModel(model);
+
+  var toBeUpdatedList = [];
+  var data = {};
+
+  self.all(model, {where: {}}, {raw: true}, function(err, docs) {
+    if (err) return cb(err);
+    async.each(docs, function(doc, cb) {
+      data = _.find(datas, {_rev: doc._rev});
+      if (data) {
+        delete doc._rev;
+        _.mergeWith(doc, data, function(dest, src) { return src; });
+        toBeUpdatedList.push(doc);
+      }
```

Comment:
If we do this, it means that only items that have matching `_rev` values will be sent to the database.

While this sounds good at first, what it means is that _the user won't know which values were updated_ since they may have had some of their items dropped from the list without warning.